### PR TITLE
[MRG] Epochs browser loads data on horizontal shift.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/2556

### CODE REVIEW

**Comment by agramfort** [comment]
@teonlamont works for you?

thanks @jaeilepp for the quick fix.

**Comment by teonlamont** [comment]
works great, thanks @jaeilepp! quick unrelated question, how do you get back to original plot when you call the histogram function `(h)`. If I press `(h)` again, it just recalculates the plot. If I try `(b)`, it just remains on the histogram.

**Comment by jaeilepp** [comment]
The histogram open to a new window, doesn't it? Just close the histogram.

**Comment by teonlamont** [comment]
for me, the histogram plots in the same window/fig. i'm starting my session with `ipython --pylab=qt`.

**Comment by jaeilepp** [comment]
Yeah, that was a bug. I think it should work now.

File: mne/viz/epochs.py [hunk 1]
```diff
@@ -382,28 +382,30 @@ def plot_epochs(epochs, picks=None, scalings=None, n_epochs=20,
...
-    adjusted with - and + (or =) keys, but this depends on the backend
-    matplotlib is configured to use (e.g., mpl.use(``TkAgg``) should work).
-    Full screen mode can be to toggled with f11 key. The amount of epochs and
-    channels per view can be adjusted with home/end and page down/page up keys.
-    Butterfly plot can be toggled with ``b`` key. Right mouse click adds a
-    vertical line to the plot.
+    The arrow keys (up/down/left/right) can be used to navigate between
+    channels and epochs and the scaling can be adjusted with - and + (or =)
+    keys, but this depends on the backend matplotlib is configured to use
+    (e.g., mpl.use(``TkAgg``) should work). Full screen mode can be to toggled
```
Comment (teonlamont) [inline mne/viz/epochs.py]: can be to toggled --> can be toggled

File: mne/viz/epochs.py [hunk 2]
```diff
@@ -634,14 +636,14 @@ def _prepare_mne_browse_epochs(params, projs, n_channels, n_epochs, scalings,
...
-    data = np.zeros((params['info']['nchan'], len(times) * len(epochs.events)))
+    data = np.zeros((params['info']['nchan'], len(times) * n_epochs))
 
     ylim = (25., 0.)  # Hardcoded 25 because butterfly has max 5 rows (5*5=25).
     # make shells for plotting traces
     offset = ylim[0] / n_channels
     offsets = np.arange(n_channels) * offset + (offset / 2.)
 
-    times = np.arange(len(data[0]))
+    times = np.arange(len(times) * len(epochs.events))
```
Comment (teonlamont) [inline mne/viz/epochs.py]: should this be n_epochs as well?

**Comment by teonlamont** [comment]
works great. thanks @jaeilepp!

**Comment by agramfort** [comment]
@teonlamont merge when you're happy

thanks @jaeilepp

File: mne/viz/epochs.py (hunk 2 above)
Comment (jaeilepp) [inline mne/viz/epochs.py]: No, ``times`` is the length of the whole data whereas ``n_epochs`` is the viewport size.

**Comment by teonlamont** [comment]
thanks @jaeilepp

**Pull request merged**